controllers/ovndbcluster_controller.go

```diff
@@ -566,21 +592,77 @@ func (r *OVNDBClusterReconciler) reconcileServices(
 	)
 	if err == nil && len(svcList.Items) > int(*(instance.Spec.Replicas)) {
 		for i := len(svcList.Items) - 1; i >= int(*(instance.Spec.Replicas)); i-- {
+			fullServiceName := fmt.Sprintf("%s-%d", serviceName, i)
 			svcLabels := map[string]string{
 				common.AppSelector:                   serviceName,
-				"statefulset.kubernetes.io/pod-name": serviceName + fmt.Sprintf("-%d", i),
+				"statefulset.kubernetes.io/pod-name": fullServiceName,
 			}
 			err = service.DeleteServicesWithLabel(
 				ctx,
 				helper,
 				instance,
 				svcLabels,
 			)
+			if err != nil {
+				err = fmt.Errorf("Error while deleting service with name %s: %w", serviceName, err)
+				return ctrl.Result{}, err
+			}
+			// Delete DNS records for deleted services/pods
+			namespace := helper.GetBeforeObject().GetNamespace()
+			err = deleteDNSData(ctx, helper, fullServiceName, namespace)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+	}
+
+	podList, err = ovndbcluster.OVNDBPods(ctx, instance, helper, serviceLabels)
+	if err != nil {
+		err = fmt.Errorf("Error while retrieving pods on %s cluster: %w", instance.Spec.DBType, err)
+		return ctrl.Result{}, err
+	}
+
+	var svc *corev1.Service
+
+	// When the cluster is attached to an external network, create DNS record for every
+	// cluster member so it can be resolved from outside cluster (edpm nodes)
+	if instance.Spec.NetworkAttachment != "" {
+		for _, ovnPod := range podList.Items[:*(instance.Spec.Replicas)] {
+			svc, err = service.GetServiceWithName(
+				ctx,
+				helper,
+				ovnPod.Name,
+				ovnPod.Namespace,
+			)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+
+			// Currently only IPv4 is supported
+			dnsIP, err := getPodIPv4InNetwork(ovnPod, instance.Namespace, instance.Spec.NetworkAttachment)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+
+			// Create DNSData CR
+			err = ovndbcluster.DnsData(
+				ctx,
+				helper,
+				svc,
+				dnsIP,
```

Comment:
This needs to be fixed as not handling ips from multiple replica, just the ip of last pod will persist.

Reply:
Thanks, reading it again got it as those created with podname. But looks unnecessary to maintain those extra dnsdata per pod. If it could be avoid it's good else can also cleanup later if others are ok.